sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py

```diff
@@ -583,36 +595,73 @@ async def set_read_only(
+        if read_only:
+            key_value = await self._impl.put_lock(
+                key=configuration_setting.key,
+                label=configuration_setting.label,
+                if_match=prep_if_match(configuration_setting.etag, match_condition),
+                if_none_match=prep_if_none_match(configuration_setting.etag, match_condition),
+                error_map=error_map,
+                **kwargs,
+            )
+        else:
+            key_value = await self._impl.delete_lock(
+                key=configuration_setting.key,
+                label=configuration_setting.label,
+                if_match=prep_if_match(configuration_setting.etag, match_condition),
+                if_none_match=prep_if_none_match(configuration_setting.etag, match_condition),
+                error_map=error_map,
+                **kwargs,
+            )
+        return ConfigurationSetting._from_generated(key_value)
+
+    @distributed_trace
+    def list_labels(
+        self,
+        *,
+        name: Optional[str] = None,
+        after: Optional[str] = None,
+        accept_datetime: Optional[Union[datetime, str]] = None,
+        fields: Optional[List[Union[str, LabelFields]]] = None,
+        **kwargs,
+    ) -> AsyncItemPaged[ConfigurationSettingLabel]:
+        """Gets a list of labels.
+
+        :keyword name: A filter for the name of the returned labels. '*' can be used as wildcard
+            in the beginning or end of the filter. For more information about supported filters, see
+            https://learn.microsoft.com/azure/azure-app-configuration/rest-api-labels?pivots=v23-11#supported-filters.
+        :paramtype name: str or None
+        :keyword after: Instructs the server to return elements that appear after the element referred to
+            by the specified token.
+        :paramtype after: str or None
+        :keyword accept_datetime: Requests the server to respond with the state of the resource at the
+            specified time.
+        :paramtype accept_datetime: ~datetime.datetime or str or None
+        :keyword fields: Specify which fields to include in the results. If not specified, will include all fields.
+            Available fields see :class:`~azure.appconfiguration.LabelFields`.
+        :paramtype fields: list[str] or list[~azure.appconfiguration.LabelFields] or None
+        :return: An async iterator of labels.
+        :rtype: ~azure.core.paging.AsyncItemPaged[~azure.appconfiguration.ConfigurationSettingLabel]
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+        """
+        if isinstance(accept_datetime, datetime):
+            accept_datetime = str(accept_datetime)
+        return self._impl.get_labels(  # type: ignore[return-value]
+            name=name,
+            after=after,
+            accept_datetime=accept_datetime,
+            select=fields,
+            cls=lambda objs: [ConfigurationSettingLabel(x.name) for x in objs],
```

Comment:
The `name` param is keyword-only.
```suggestion
            cls=lambda objs: [ConfigurationSettingLabel(name=x.name) for x in objs],
```